server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputerTests.java

```diff
@@ -616,42 +621,57 @@ public void testComputeConsideringShardSizes() {
 
         var discoveryNodesBuilder = DiscoveryNodes.builder()
             .add(createDiscoveryNode("node-0", DiscoveryNodeRole.roles()))
-            .add(createDiscoveryNode("node-1", DiscoveryNodeRole.roles()));
+            .add(createDiscoveryNode("node-1", DiscoveryNodeRole.roles()))
+            .add(createDiscoveryNode("node-2", DiscoveryNodeRole.roles()));
 
         var metadataBuilder = Metadata.builder();
         var routingTableBuilder = RoutingTable.builder();
 
-        for (int i = 0; i < 10; i++) {
+        var commonIndexSettings = Settings.builder()
+            .put("index.number_of_shards", 1)
+            .put("index.number_of_replicas", 0)
+            .put("index.version.created", Version.CURRENT)
+            .put("index.routing.allocation.exclude._name", "node-2")
+            .build();
+
+        ShardRouting index0PrimaryShard;
+        ShardRouting index0ReplicaShard;
+        {
+            var indexName = "index-0";
+
+            metadataBuilder.put(IndexMetadata.builder(indexName).settings(commonIndexSettings));
+
+            var indexId = metadataBuilder.get(indexName).getIndex();
+            var shardId = new ShardId(indexId, 0);
+
+            index0PrimaryShard = newShardRouting(shardId, "node-1", null, true, STARTED);
+            index0ReplicaShard = switch (randomIntBetween(0, 4)) {
+                // shard is started on the desired node
+                case 0 -> newShardRouting(shardId, "node-0", null, false, STARTED);
```

Comment:
These cases are all good, but I wonder why we do not have:

1. relocating to an undesired node.
2. initializing on an undesired node.

Reply:
Existing scenarios were covering the branches that involved disk usage simulation, but I added suggested ones just in case.